edsnlp/matchers/utils/text.py

```diff
@@ -32,15 +35,18 @@ def get_text(
 
     attr = attr.upper()
 
-    if not ignore_excluded:
+    if not ignore_excluded and not ignore_space_tokens:
         if attr == "TEXT":
             return doclike.text
         elif attr == "LOWER":
             return doclike.text.lower()
         else:
             tokens = doclike
     else:
-        tokens = [t for t in doclike if t.tag_ != "EXCLUDED"]
+        if not ignore_space_tokens:
+            tokens = [t for t in doclike if t.tag_ != "EXCLUDED"]
```

Comment:
it should be == ?
```suggestion
            tokens = [t for t in doclike if t.tag_ == "EXCLUDED"]
```

Reply:
Here we're only interested in non-excluded tokens, so `!=` is relevant